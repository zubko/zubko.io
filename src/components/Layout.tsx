/**
 * Root component of the page
 */

import { graphql, StaticQuery } from 'gatsby';
import { ReactNode } from 'react';

import { Footer, Header } from '../components';
import { rhythm } from '../Typography';

type Props = {
  children: ReactNode;
};

const Layout = ({ children }: Props) => (
  <StaticQuery
    query={graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            title
          }
        }
      }
    `}
    render={data => {
      return (
        <div
          css={{
            margin: `0 auto`,
            maxWidth: rhythm(24),
            padding: `0px 1.0875rem 1.45rem`,
            paddingTop: 0,
          }}>
          <header>
            <Header siteTitle={data.site.siteMetadata.title} />
          </header>
          <main>{children}</main>
          <Footer />
        </div>
      );
    }}
  />
);

export default Layout;
