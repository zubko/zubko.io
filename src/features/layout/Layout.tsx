/**
 * Root component of the page
 */

import { graphql, StaticQuery } from "gatsby";
import { ReactNode } from "react";
import styled from "styled-components";

import { rhythm } from "../../Typography";
import { Footer } from "./Footer";
import { Header } from "./Header";

type Props = {
  children: ReactNode;
};

export const Layout = ({ children }: Props) => (
  <StaticQuery
    query={graphql`
      query SiteTitleQuery {
        site {
          siteMetadata {
            title
          }
        }
      }
    `}
    render={(data) => {
      return (
        <Container>
          <header>
            <Header siteTitle={data.site.siteMetadata.title} />
          </header>
          <main>{children}</main>
          <Footer />
        </Container>
      );
    }}
  />
);

const Container = styled.div`
  margin: 0 auto;
  max-width: ${rhythm(24)};
  padding: 0px 1.0875rem 1.45rem;
  padding-top: 0px;
`;
