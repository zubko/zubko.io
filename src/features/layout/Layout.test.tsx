import { StaticQuery } from "gatsby";
import renderer from "react-test-renderer";

import { Layout } from "./Layout";

beforeEach(() => {
  (StaticQuery as jest.Mock<StaticQuery>)
    .mockImplementationOnce(({ render }) =>
      render({
        site: {
          siteMetadata: {
            title: "title"
          }
        }
      })
    )
    .mockImplementationOnce(({ render }) =>
      render({
        allMenuYaml: {
          edges: [
            {
              node: {
                path: "/path-1/",
                title: "title-1"
              }
            },
            {
              node: {
                path: "/path-2/",
                title: "title-2"
              }
            }
          ]
        }
      })
    );
});

describe("Layout", () => {
  it("renders without regressions", () => {
    const tree = renderer.create(<Layout>Child</Layout>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
