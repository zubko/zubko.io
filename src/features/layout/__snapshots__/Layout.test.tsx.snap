// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders without regressions 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1argzsk{margin:0 auto;max-width:38.832rem;padding:0px 1.0875rem 1.45rem;padding-top:0px;}",
      }
    }
    data-emotion="css 1argzsk"
  />,
  <div
    className="css-1argzsk"
  >
    <header>
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-onf4p3{margin-bottom:0.809rem;}",
          }
        }
        data-emotion="css onf4p3"
      />
      <div
        className="css-onf4p3"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-vls6ey{padding-top:0.809rem;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-webkit-justify-content:space-between;justify-content:space-between;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
            }
          }
          data-emotion="css vls6ey"
        />
        <div
          className="css-vls6ey"
        >
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": ".css-1crlzb{-webkit-text-decoration:none;text-decoration:none;background:none;text-shadow:none;}.css-1crlzb:hover{-webkit-text-decoration:none;text-decoration:none;}.css-1crlzb >h1{color:#2C8127;margin:0;font-weight:600;font-size:1.21225rem;}",
              }
            }
            data-emotion="css 1crlzb"
          />
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": ".css-vbatjp-Link{color:#2C8127;-webkit-text-decoration:none;text-decoration:none;background-image:linear-gradient(
    to top,
    rgba(0, 0, 0, 0),
    rgba(0, 0, 0, 0) 1px,
    currentColor 1px,
    currentColor 2px,
    rgba(0, 0, 0, 0) 2px
  );-webkit-text-decoration:none;text-decoration:none;background:none;text-shadow:none;}.css-vbatjp-Link:hover{background-image:none;-webkit-filter:brightness(1.3);filter:brightness(1.3);}.css-vbatjp-Link:hover{-webkit-text-decoration:none;text-decoration:none;}.css-vbatjp-Link >h1{color:#2C8127;margin:0;font-weight:600;font-size:1.21225rem;}",
              }
            }
            data-emotion="css vbatjp-Link"
          />
          <a
            className="css-vbatjp-Link"
            href="/"
          >
            <h1>
              title
            </h1>
          </a>
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": "@media (max-width: 779px){.css-12od4kl{display:none;}}",
              }
            }
            data-emotion="css 12od4kl"
          />
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": ".css-8d8lv6-MenuHorizontal{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;}@media (max-width: 779px){.css-8d8lv6-MenuHorizontal{display:none;}}",
              }
            }
            data-emotion="css 8d8lv6-MenuHorizontal"
          />
          <div
            className="css-8d8lv6-MenuHorizontal"
          >
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": ".css-1be2rwe{-webkit-text-decoration:none;text-decoration:none;fontsize:1rem;font-weight:medium;background:none;}",
                }
              }
              data-emotion="css 1be2rwe"
            />
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": ".css-2w87gj-Link{color:#2C8127;-webkit-text-decoration:none;text-decoration:none;background-image:linear-gradient(
    to top,
    rgba(0, 0, 0, 0),
    rgba(0, 0, 0, 0) 1px,
    currentColor 1px,
    currentColor 2px,
    rgba(0, 0, 0, 0) 2px
  );-webkit-text-decoration:none;text-decoration:none;fontsize:1rem;font-weight:medium;background:none;}.css-2w87gj-Link:hover{background-image:none;-webkit-filter:brightness(1.3);filter:brightness(1.3);}",
                }
              }
              data-emotion="css 2w87gj-Link"
            />
            <a
              className="css-2w87gj-Link"
              href="/path-1/"
            >
              title-1
            </a>
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": ".css-11b9gp{color:white;margin-left:0.5rem;margin-right:0.5rem;cursor:default;}",
                }
              }
              data-emotion="css 11b9gp"
            />
            <span
              className="css-11b9gp"
            />
            <a
              className="css-2w87gj-Link"
              href="/path-2/"
            >
              title-2
            </a>
          </div>
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": "@media (min-width: 780px){.css-1u0qo0x{display:none;}}",
              }
            }
            data-emotion="css 1u0qo0x"
          />
          <div
            className="css-1u0qo0x"
          >
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": ".css-1dridds{cursor:pointer;color:#2C8127;border:1px solid #2C8127;padding:0.5rem;background-color:transparent;}.css-1dridds:focus{outline:0;}",
                }
              }
              data-emotion="css 1dridds"
            />
            <button
              className="css-1dridds"
              onClick={[Function]}
            >
              Menu
            </button>
          </div>
        </div>
        <div
          className="css-1u0qo0x"
        >
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": ".css-lnkmq4{display:none;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding-bottom:0.809rem;padding-top:1.618rem;}",
              }
            }
            data-emotion="css lnkmq4"
          />
          <div
            className="css-lnkmq4"
          >
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": ".css-u6unsj{-webkit-text-decoration:none;text-decoration:none;font-size:1rem;font-weight:medium;background:none;}",
                }
              }
              data-emotion="css u6unsj"
            />
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": ".css-wj5h1p-Link{color:#2C8127;-webkit-text-decoration:none;text-decoration:none;background-image:linear-gradient(
    to top,
    rgba(0, 0, 0, 0),
    rgba(0, 0, 0, 0) 1px,
    currentColor 1px,
    currentColor 2px,
    rgba(0, 0, 0, 0) 2px
  );-webkit-text-decoration:none;text-decoration:none;font-size:1rem;font-weight:medium;background:none;}.css-wj5h1p-Link:hover{background-image:none;-webkit-filter:brightness(1.3);filter:brightness(1.3);}",
                }
              }
              data-emotion="css wj5h1p-Link"
            />
            <a
              className="css-wj5h1p-Link"
              href="/path-1/"
            >
              title-1
            </a>
            <style
              dangerouslySetInnerHTML={
                Object {
                  "__html": ".css-1xtjhx3{height:0.5rem;color:white;cursor:default;}",
                }
              }
              data-emotion="css 1xtjhx3"
            />
            <span
              className="css-1xtjhx3"
            />
            <a
              className="css-wj5h1p-Link"
              href="/path-2/"
            >
              title-2
            </a>
          </div>
        </div>
      </div>
    </header>
    <main>
      Child
    </main>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-1ckf70n-Footer{margin-top:1.618rem;}",
        }
      }
      data-emotion="css 1ckf70n-Footer"
    />
    <footer
      className="css-1ckf70n-Footer"
    >
      <p>
        ~~~
        <br />
         Â© 
        2022
        , Built
         
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-de545j{-webkit-text-decoration:none;text-decoration:none;background:none;}",
            }
          }
          data-emotion="css de545j"
        />
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-100imn2-Link{color:#2C8127;-webkit-text-decoration:none;text-decoration:none;background-image:linear-gradient(
    to top,
    rgba(0, 0, 0, 0),
    rgba(0, 0, 0, 0) 1px,
    currentColor 1px,
    currentColor 2px,
    rgba(0, 0, 0, 0) 2px
  );-webkit-text-decoration:none;text-decoration:none;background:none;}.css-100imn2-Link:hover{background-image:none;-webkit-filter:brightness(1.3);filter:brightness(1.3);}",
            }
          }
          data-emotion="css 100imn2-Link"
        />
        <a
          className="css-100imn2-Link"
          href="/acknowledgements"
        >
          with ðŸ’œ and open source
           
          <style
            dangerouslySetInnerHTML={
              Object {
                "__html": ".css-1fw2vfc{color:violet;}",
              }
            }
            data-emotion="css 1fw2vfc"
          />
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-react fa-lg  css-1fw2vfc"
            data-icon="react"
            data-prefix="fab"
            focusable="false"
            height={25}
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            width={25}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M418.2 177.2c-5.4-1.8-10.8-3.5-16.2-5.1.9-3.7 1.7-7.4 2.5-11.1 12.3-59.6 4.2-107.5-23.1-123.3-26.3-15.1-69.2.6-112.6 38.4-4.3 3.7-8.5 7.6-12.5 11.5-2.7-2.6-5.5-5.2-8.3-7.7-45.5-40.4-91.1-57.4-118.4-41.5-26.2 15.2-34 60.3-23 116.7 1.1 5.6 2.3 11.1 3.7 16.7-6.4 1.8-12.7 3.8-18.6 5.9C38.3 196.2 0 225.4 0 255.6c0 31.2 40.8 62.5 96.3 81.5 4.5 1.5 9 3 13.6 4.3-1.5 6-2.8 11.9-4 18-10.5 55.5-2.3 99.5 23.9 114.6 27 15.6 72.4-.4 116.6-39.1 3.5-3.1 7-6.3 10.5-9.7 4.4 4.3 9 8.4 13.6 12.4 42.8 36.8 85.1 51.7 111.2 36.6 27-15.6 35.8-62.9 24.4-120.5-.9-4.4-1.9-8.9-3-13.5 3.2-.9 6.3-1.9 9.4-2.9 57.7-19.1 99.5-50 99.5-81.7 0-30.3-39.4-59.7-93.8-78.4zM282.9 92.3c37.2-32.4 71.9-45.1 87.7-36 16.9 9.7 23.4 48.9 12.8 100.4-.7 3.4-1.4 6.7-2.3 10-22.2-5-44.7-8.6-67.3-10.6-13-18.6-27.2-36.4-42.6-53.1 3.9-3.7 7.7-7.2 11.7-10.7zM167.2 307.5c5.1 8.7 10.3 17.4 15.8 25.9-15.6-1.7-31.1-4.2-46.4-7.5 4.4-14.4 9.9-29.3 16.3-44.5 4.6 8.8 9.3 17.5 14.3 26.1zm-30.3-120.3c14.4-3.2 29.7-5.8 45.6-7.8-5.3 8.3-10.5 16.8-15.4 25.4-4.9 8.5-9.7 17.2-14.2 26-6.3-14.9-11.6-29.5-16-43.6zm27.4 68.9c6.6-13.8 13.8-27.3 21.4-40.6s15.8-26.2 24.4-38.9c15-1.1 30.3-1.7 45.9-1.7s31 .6 45.9 1.7c8.5 12.6 16.6 25.5 24.3 38.7s14.9 26.7 21.7 40.4c-6.7 13.8-13.9 27.4-21.6 40.8-7.6 13.3-15.7 26.2-24.2 39-14.9 1.1-30.4 1.6-46.1 1.6s-30.9-.5-45.6-1.4c-8.7-12.7-16.9-25.7-24.6-39s-14.8-26.8-21.5-40.6zm180.6 51.2c5.1-8.8 9.9-17.7 14.6-26.7 6.4 14.5 12 29.2 16.9 44.3-15.5 3.5-31.2 6.2-47 8 5.4-8.4 10.5-17 15.5-25.6zm14.4-76.5c-4.7-8.8-9.5-17.6-14.5-26.2-4.9-8.5-10-16.9-15.3-25.2 16.1 2 31.5 4.7 45.9 8-4.6 14.8-10 29.2-16.1 43.4zM256.2 118.3c10.5 11.4 20.4 23.4 29.6 35.8-19.8-.9-39.7-.9-59.5 0 9.8-12.9 19.9-24.9 29.9-35.8zM140.2 57c16.8-9.8 54.1 4.2 93.4 39 2.5 2.2 5 4.6 7.6 7-15.5 16.7-29.8 34.5-42.9 53.1-22.6 2-45 5.5-67.2 10.4-1.3-5.1-2.4-10.3-3.5-15.5-9.4-48.4-3.2-84.9 12.6-94zm-24.5 263.6c-4.2-1.2-8.3-2.5-12.4-3.9-21.3-6.7-45.5-17.3-63-31.2-10.1-7-16.9-17.8-18.8-29.9 0-18.3 31.6-41.7 77.2-57.6 5.7-2 11.5-3.8 17.3-5.5 6.8 21.7 15 43 24.5 63.6-9.6 20.9-17.9 42.5-24.8 64.5zm116.6 98c-16.5 15.1-35.6 27.1-56.4 35.3-11.1 5.3-23.9 5.8-35.3 1.3-15.9-9.2-22.5-44.5-13.5-92 1.1-5.6 2.3-11.2 3.7-16.7 22.4 4.8 45 8.1 67.9 9.8 13.2 18.7 27.7 36.6 43.2 53.4-3.2 3.1-6.4 6.1-9.6 8.9zm24.5-24.3c-10.2-11-20.4-23.2-30.3-36.3 9.6.4 19.5.6 29.5.6 10.3 0 20.4-.2 30.4-.7-9.2 12.7-19.1 24.8-29.6 36.4zm130.7 30c-.9 12.2-6.9 23.6-16.5 31.3-15.9 9.2-49.8-2.8-86.4-34.2-4.2-3.6-8.4-7.5-12.7-11.5 15.3-16.9 29.4-34.8 42.2-53.6 22.9-1.9 45.7-5.4 68.2-10.5 1 4.1 1.9 8.2 2.7 12.2 4.9 21.6 5.7 44.1 2.5 66.3zm18.2-107.5c-2.8.9-5.6 1.8-8.5 2.6-7-21.8-15.6-43.1-25.5-63.8 9.6-20.4 17.7-41.4 24.5-62.9 5.2 1.5 10.2 3.1 15 4.7 46.6 16 79.3 39.8 79.3 58 0 19.6-34.9 44.9-84.8 61.4zm-149.7-15c25.3 0 45.8-20.5 45.8-45.8s-20.5-45.8-45.8-45.8c-25.3 0-45.8 20.5-45.8 45.8s20.5 45.8 45.8 45.8z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </p>
    </footer>
  </div>,
]
`;
